models/agent_helpers.go

```diff
@@ -637,6 +637,63 @@ type CreateAgentParams struct {
 	DisableCollectors              []string
 }
 
+func verifyNodeAndAgent(nodeType NodeType, agentType AgentType) error {
```

Comment:
I think we can make these two functions return just boolean and return error in `CreateAgent`, in that case code will be simpler